pkg/proxy/backend/backend_conn_mgr.go

```diff
@@ -160,11 +160,12 @@ func (mgr *BackendConnManager) getBackendIO(ctx ConnContext, auth *Authenticator
 	}
 	// wait for initialize
 	bctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	selector := r.Route()
 	addr, err := backoff.RetryNotifyWithData(
 		func() (string, error) {
-			addr, err := r.Route(mgr)
-			if !errors.Is(err, router.ErrNoInstanceToSelect) {
-				return addr, backoff.Permanent(err)
+			addr := selector.Next()
+			if len(addr) == 0 {
+				return "", router.ErrNoInstanceToSelect
```

Comment:
Just a note, `backoff.Permanent` will stop retrying/backoffing. Normal errors won't stop the loop.

Reply:
I know. The only error it may return is no instances to route, and it should continue to try.